tls/s2n_resume.c

```diff
@@ -610,28 +610,30 @@ int s2n_connection_is_ocsp_stapled(struct s2n_connection *conn)
     }
 }
 
-int s2n_config_is_encrypt_decrypt_key_available(struct s2n_config *config)
+S2N_RESULT s2n_config_is_encrypt_key_available(struct s2n_config *config)
 {
+    RESULT_ENSURE_REF(config);
+
     uint64_t now = 0;
     struct s2n_ticket_key *ticket_key = NULL;
-    POSIX_GUARD_RESULT(s2n_config_wall_clock(config, &now));
-    POSIX_ENSURE_REF(config->ticket_keys);
+    RESULT_GUARD(s2n_config_wall_clock(config, &now));
+    RESULT_ENSURE_REF(config->ticket_keys);
 
     uint32_t ticket_keys_len = 0;
-    POSIX_GUARD_RESULT(s2n_set_len(config->ticket_keys, &ticket_keys_len));
+    RESULT_GUARD(s2n_set_len(config->ticket_keys, &ticket_keys_len));
 
     for (uint32_t i = ticket_keys_len; i > 0; i--) {
         uint32_t idx = i - 1;
-        POSIX_GUARD_RESULT(s2n_set_get(config->ticket_keys, idx, (void **) &ticket_key));
+        RESULT_GUARD(s2n_set_get(config->ticket_keys, idx, (void **) &ticket_key));
         uint64_t key_intro_time = ticket_key->intro_timestamp;
 
         if (key_intro_time < now
                 && now < key_intro_time + config->encrypt_decrypt_key_lifetime_in_nanos) {
-            return 1;
+            return S2N_RESULT_OK;
         }
     }
 
-    return 0;
+    RESULT_BAIL(S2N_ERR_KEY_USED_IN_SESSION_TICKET_NOT_FOUND);
```

Comment:
This isn't the right error. The message doesn't match the situation:
https://github.com/aws/s2n-tls/blob/1ed5624c24c80b5ef93f93a72ee93b503ddd6eba/error/s2n_errno.c#L237
How about "S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY" instead?

Reply:
Got it. I didn't realize the error situation was described in this file. Thank you!